src/response/response.rs

```diff
@@ -31,41 +35,38 @@ impl Response {
     #[inline]
     pub fn map<F>(self, map: F) -> Self
     where
-        F: FnOnce(HttpBody<ResponseBody>) -> HttpBody<ResponseBody>,
+        F: FnOnce(Either<BufferBody, BoxBody>) -> Either<BufferBody, BoxBody>,
     {
         Self {
-            inner: self.inner.map(map),
+            inner: self.inner.map(|body| ResponseBody {
```

Comment:
It's a good thing the response body isn't polled until it's unwrapped from `Response` on line 87 in `crate::app::service`.